pkg/service/v0/service.go

```diff
@@ -303,3 +339,59 @@ func (p WopiServer) stat(path, auth string) (*provider.StatResponse, error) {
 	}
 	return rsp, nil
 }
+
+func getUserAndAuthToken(r *http.Request, tm config.TokenManager) (username, revaToken string, err error) {
+
+	ctx := r.Context()
+
+	tokenManager, err := revajwt.New(map[string]interface{}{
+		"secret":  tm.JWTSecret,
+		"expires": tm.TokenTTL.Seconds(),
+	})
+	if err != nil {
+		return "", "", err
+	}
+
+	user := revauser.ContextMustGetUser(ctx)
+	scope, err := scope.GetOwnerScope()
+	if err != nil {
+		return "", "", err
+	}
+
+	revaToken, err = tokenManager.MintToken(ctx, user, scope)
+	if err != nil {
+		return "", "", err
+	}
+
+	username = user.DisplayName
+
+	// TODO: if CS3org WOPI server mints the final REVA JWT secret,
```

Comment:
Do we want to always mint a token like we're doing here, or also provide support for when a token is present in the request, because it was minted by the proxy? (I think this is what you're trying to say with this comment) 🤓 I'm okay with either, but I'd move this comment to a follow up issue in the repo and keep the codebase a bit cleaner 👍; analyzing the commented code, it is going to work the same way.

Reply:
Ultimately we need to find a cleaner way to mint tokens. Services minting things involves coordinated shared secrets, which we don't currently have. I'll bring this to a broader audience #soon

Reply:
Biggest Problem in that scenario is, that the token in the user context is only valid for a few minutes, but you probably would like to edit your document opened in Collabora for a longer time. Therefore we need to mint a token which has a way longer validity.

Reply:
ah, so the logic is more convoluted, I see 🤔